openedx2zim/annex.py

```diff
@@ -1,136 +1,134 @@
 import re
-import os
 import html
 import uuid
 import json
-import urllib
 import pathlib
 import collections
 
 from bs4 import BeautifulSoup
 
 from .constants import getLogger
-from .utils import jinja, markdown, prepare_url
+from .utils import jinja, markdown
 
 logger = getLogger()
 
 
-def forum(instance_connection, scraper):
-    forum_output = scraper.build_dir.joinpath("forum")
-    forum_output.mkdir(parents=True, exist_ok=True)
-    content = instance_connection.get_page(
-        scraper.instance_url + "/courses/" + scraper.course_id + "/discussion/forum"
-    )
-    good_content = BeautifulSoup(content, "lxml").find(
-        "script", attrs={"id": "thread-list-template"}
-    )
-    category = collections.OrderedDict()
-    if good_content:
-        soup = BeautifulSoup(content, "lxml")
+class MoocForum:
+    def __init__(self, scraper):
+        self.scraper = scraper
+        self.threads = []
+        self.categories = collections.OrderedDict()
+        self.staff_user = []
+        self.output_path = self.scraper.build_dir.joinpath("forum")
+        self.output_path.mkdir(parents=True, exist_ok=True)
 
-        all_category = soup.find_all(
-            "li", attrs={"class": "forum-nav-browse-menu-item"}
-        )
-        if len(all_category) == 0:
-            soup = BeautifulSoup(
-                good_content.text, "lxml"
-            )  # On Fun plateform, categorie list is in script with id thread-list-template
-            all_category = soup.find_all(
-                "li", attrs={"class": "forum-nav-browse-menu-item"}
-            )
-        for cat in all_category:
+    def add_categories(self, categories):
+        for category in categories:
             if (
-                cat.has_attr("id")
-                and cat["id"] in ["all_discussions", "posts_following"]
+                category.has_attr("id")
```

Comment:
Why not testing `category.get("id")` ?

Reply:
Am, yup, that'd be better, I overlooked that it seems. However, will do that the next time I push code.